### ISSUE TITLE
The nn can't converge while solving my odes

### ISSUE THREAD

**Issue by Edward-CY** [lululxvi/deepxde#385]
Hi, lu
Thanks for your great work!
I am learning how to solve my odes with initial conditions. It is a complex odes that describe a chemical process. I reference to the example named **Lotka_Volterra_Colab**. It seems that the nn can't convergence while solving my odes. My code is as follows:

> from __future__ import absolute_import
> from __future__ import division
> from __future__ import print_function
> import math
> from deepxde.backend import tf
> import numpy as np
> import matplotlib.pyplot as plt
> import deepxde as dde
> 
> def ode_system(x, y):
>     # Temperature in CSTR (K)
>     T = y[:, 0:1]
>     #Concentration of A in CSTR (mol/m^3)
>     Ca = y[:, 1:2]
>     dT_t = dde.grad.jacobian(y, x, i=0)
>     dCa_t = dde.grad.jacobian(y, x, i=1)
>     
>     Tc = 280  # nominal = 300
>     # Volumetric Flowrate (m^3/sec)
>     q = 100   # nominal = 100
>     # Feed Concentration (mol/m^3)
>     Caf = 1 # nominal = 1
>     # Feed Temperature (K)
>     Tf = 350  # nominal = 350
>     
>     #Ca_ss = 0.87725294608097;
>     #T_ss = 324.475443431599;
> 
>     # Volume of CSTR (m^3)
>     V = 100
>     # Density of A-B Mixture (kg/m^3)
>     rho = 1000
>     # Heat capacity of A-B Mixture (J/kg-K)
>     Cp = 0.239
>     # Heat of reaction for A->B (J/mol)
>     mdelH = 5e4
>     # E - Activation energy in the Arrhenius Equation (J/mol)
>     # R - Universal Gas Constant = 8.31451 J/mol-K
>     EoverR = 8750
>     # Pre-exponential factor (1/sec)
>     k0 = 7.2e10    
>     # U - Overall Heat Transfer Coefficient (W/m^2-K)
>     # A - Area - this value is specific for the U calculation (m^2)
>     UA = 5e4
> 
>     return [
>         dT_t - q/V*(Tf - T) + mdelH/(rho*Cp)*k0*tf.math.exp(-EoverR/T)*Ca + UA/V/rho/Cp*(Tc-T),
>         dCa_t - q/V*(Caf - Ca) - k0*tf.math.exp(-EoverR/T)*Ca]
> 
> 
> def boundary(_, on_initial):
>     return on_initial
> 
> geom = dde.geometry.TimeDomain(0.0, 10.0)
> ic1 = dde.IC(geom, lambda X: 324.475443431599, boundary, component=0)
> ic2 = dde.IC(geom, lambda X: 0.87725294608097, boundary, component=1)
> data = dde.data.PDE(geom, ode_system, [ic1, ic2], 3000, 2, num_test=3000)
> 
> layer_size = [1] + [64] * 3 + [2]
> activation = "tanh"
> initializer = "Glorot normal"
> net = dde.maps.FNN(layer_size, activation, initializer)
> 
> model = dde.Model(data, net)
> model.compile("adam", lr=0.001)
> losshistory, train_state = model.train(epochs=30000)
> 
> model.compile("L-BFGS")
> losshistory, train_state = model.train()
> 
> dde.saveplot(losshistory, train_state, issave=True, isplot=True)

After it runs. It shows:

> Step      Train loss                                  Test loss                                   Test metric
> 0         [inf, inf, 1.05e+05, 7.70e-01]              [inf, inf, 1.05e+05, 7.70e-01]              []  
> 1000      [nan, nan, nan, nan]                        [nan, nan, nan, nan]                        []  
> 2000      [nan, nan, nan, nan]                        [nan, nan, nan, nan]                        []  
> 3000      [nan, nan, nan, nan]                        [nan, nan, nan, nan]                        []  

I have tried many methods like changing lr or activation function. I wonder what the problem is. Thank you!

**Comment by SanjuSoni** [comment]
I think you have some bug in jacobian definition:

dT_t = dde.grad.jacobian(y, x, i=0)---------------------------> dde.grad.jacobian(y, x, i=0, j= 0)
dCa_t = dde.grad.jacobian(y, x, i=1)------------------------->dde.grad.jacobian(y, x, i = 1, j=0)


try it.
